docs/DeveloperGuide.md

```diff
@@ -1,12 +1,251 @@
+
+<div align="justify">  
+The class diagram as depicted below shows the behaviour of the passenger delete operation explained in two parts.  
+This scenario assumes that the manager have input a valid `passenger delete` command for easier readability.  
+</div>  
+
+
+Part 1: 
+* The valid `passenger delete` command is passed as a `String lineInput` into the function `parse` to valid 
+the type of command that needs to be executed.
+* Once validated, a new Command called `DeletePassengerCommand` should be instantiated.
+
+![class diagram](../docs/ug-diagrams/images/passengerdltcmdClassDiagram_0.jpg)
+
+Part 2:
+* `DeletePassengerCommand` would then call `execute(passengers, lineInput)`, 
+associating itself with PassengerList to obtain the existing `ArrayList<FlightInfo> passengers`.
+* From there, `deleteOperation` method would run within method `execute` and terminate only when there is no such passenger
+found or when a passenger is removed.
+
+
+![class diagram](../docs/ug-diagrams/images/passengerdltcmdClassDiagram_1.jpg)
+
+**Sequence diagram**
+
+![sequence diagram](../docs/ug-diagrams/images/passengerDeleteCmdSeqDiagram.jpg)
+
+When `paser` verifies that the command is an entity `passenger` and `delete` operation,
+`DeletePassengerCommend` is instantiated.  
+1. `execute(passenger, lineInput)` will run within `DeletePassengerCommend` 
+which in turn would call on `deleteOperation(passengerDetail)` in `PassengerList` class.
+2. Within `deleteOperation(passengerDetail)`, it would loop the arraylist `passengers` obtained from `OperationList`.
+3. If passenger is found, delete object `passenger` from `passengers`. ELse, return without deletion.
+
+---
+
+### List passengers feature
+
+<div align="justify">  
+
+A list function lists out all passengers' details in table form which is facilitated by `ListPassengerCommand`. It extends an abstract `Command` with an override method called `execute`.
+The abstract `Command` extends a `Parser` which holds and validates the User input to determine the type of command.
+
+</div>  
+
+Furthermore, `ListPassengerCommand` implements the following operation:
+* `execute(OperationList passengers, String lineInput)` &#8594; An override method inherited from `command` 
+to execute the list operation that should list out all the passengers from the logbook.
+
+<div align="justify">  
+The class diagram as depicted below shows the behaviour of the passenger list operation explained in two parts.  
+This scenario assumes that the manager have input a valid `passenger list` command for easier readability.  
+</div>
+
+Part 1:  
+
+* The valid `passenger list` command is passed as a `String lineInput` into the function `parse` to valid
+  the type of command that needs to be executed.
+* Once validated, a new Command called `ListPassengerCommand` should be instantiated.  
+
+![class diagram](../docs/ug-diagrams/images/passengerListcmdClassDiagram_0.jpg)  
```

Comment:
I see {abstract} at the bottom of Command class. Not sure if it is supposed to be there.